### DISCUSSION TITLE
Is it possible to use it as general purpose code sharing  solution?

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [hyperledger-labs/solang#1571] [Q&A]
I need to share code with EVM, Solana, Polkadot pallets, and Cosmos Cosmwasm. 

Can I write Solidity pure functions and structures, and share code?

Structures can be shared via protobuf, but functions are not.

So can I use this as function sharing code tool?

**Comment by LucasSte** [comment]
EVM, Solana and Polkadot operate slightly different, so the compiler builtins you need to interact with each chain are also different in Solidity.

If your intention is to have pure functions only, you can pretty much use the same functions for all those three targets (Solang only targets Solana and Polkadot) and use Solang to compile for Solana and Polkadot.
